astrosat/tests/factories.py

```diff
@@ -24,14 +26,15 @@ class Meta:
 
     @factory.post_generation
     def password(self, create: bool, extracted: Sequence[Any], **kwargs):
-        password = FactoryFaker(
-            "password",
-            length=42,
-            special_chars=True,
-            digits=True,
-            upper_case=True,
-            lower_case=True,
-        ).generate(extra_kwargs={})
+
+        password_kwargs = {
+            "length": 42,
+            "special_chars": True,
+            "digits": True,
+            "upper_case": True,
+            "lower_case": True,
+        }
+        password = fake.password(**password_kwargs)
```

Comment:
Is it possible to do it like this? The `password_kwargs` object isn't used anywhere else, so having a variable for it seems wasteful. The `*` and `**` bit always gets me and I have to keep looking up what they mean again and again though. So maybe this suggestion is just stupid. I wouldn't expect this to delay the merge, more for my understanding. As most of my questions seem to be.....

```suggestion
        password = fake.password({
            length: 42,
            special_chars: True,
            digits: True,
            upper_case=True,
            lower_case=True,
        })
```

Reply:
_Almost_.  If you define them as arguments then the code would be:
```
password = fake.password(length=42, special_chars=True, digits=True, upper_case=True, lower_case=True)
```

The `**` and `*` dereference a dictionary and list respectively. So:

```
def do_stuff_with_args_and_kwargs(*args, **kwargs):
  for i, arg in enumerate(args):
    print(f"{i}: {arg}")
  for i, kwarg in enumerate(kwargs.items()):
    print(f"{i}: {kwarg[0]}={kwarg[1]}")

my_list = [1,2,3]
my_dict = {"a":"foo","b":"bar","c":"baz"}

>>> do_stuff_with_args_and_kwargs(my_list, my_dict)
0: [1, 2, 3]
1: {'a': 'foo', 'b': 'bar', 'c': 'baz'}
>>> do_stuff_with_args_and_kwargs(*my_list, my_dict)
0: 1
1: 2
2: 3
3: {'a': 'foo', 'b': 'bar', 'c': 'baz'}
>>> do_stuff_with_args_and_kwargs(*my_list, **my_dict)
0: 1
1: 2
2: 3
0: a=foo
1: b=bar
2: c=baz
```

The only reason I did it in two steps was b/c it mirrors the code in **django-astrosat-users**.

Reply:
So using **named parameters**, instead of de-structuring of a dictionary, makes sense. Most seem to use these interchangeably, just depends on the circumstances. If `**` effectively de-structures a dictionary, can you do:

```python
fake.password(**{ length: 42, digits: True });
```

I accept this might be more effort than using **named parameters**, more trying to figure out if it's possible.

Reply:
Yes.  You could do that.

But in Python-land

```
fake.password(length=42, digits=True)
```

is preferred over

```
fake.password(**{"length": 42, "digits": True})
```